### PR TITLE
 Make it possible to run badge server on App Engine.

### PR DESCRIPTION
**Pull request by brianquinlan**

### CODE REVIEW

File: badge_server/main.py [hunk 1]
```diff
@@ -339,9 +304,14 @@ def one_badge_image():
...
     color = STATUS_COLOR_MAPPING[status]
-    url = URL_PREFIX + '{}-{}-{}.svg'.format(badge_name, status, color)
 
-    response = flask.make_response(requests.get(url).text)
+    response = flask.make_response(
+        pybadges.badge(
+            left_text=badge_name,
+            right_text=status,
+            right_color=color,
+            whole_link=url_prefix + flask.url_for('one_badge_target',
```
Comment (liyanhui1228) [inline badge_server/main.py]: Could you extract the whole_link to a separate line?

File: badge_server/datastore_cache.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+from google.cloud import datastore
+
+
+class DatastoreCache:
+    def __init__(self):
+        self._datastore_client = datastore.Client()
+        
+    def get(self, name: str) -> Any:
+        """Returns a Python value given a key. None if not found."""
+
```
Comment (ylil93) [inline badge_server/datastore_cache.py]: I think this line can be deleted

File: badge_server/.gcloudignore [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#
+.gcloudignore
+# If you would like to upload your .git directory, .gitignore file or files
+# from your .gitignore file, remove the corresponding line
+# below:
+.git
+.gitignore
+
+# Python pycache:
+__pycache__/
```
[the remark this answers is not in the document]
Comment (brianquinlan) [reply badge_server/.gcloudignore]: Done

File: badge_server/main.py (hunk 1 above)
Comment (brianquinlan) [reply badge_server/main.py]: Done

File: badge_server/datastore_cache.py (hunk 2 above)
Comment (brianquinlan) [reply badge_server/datastore_cache.py]: Done

**Comment by liyanhui1228** [comment]
Is it safe for us to remove the GKE version badge server and the memorystore stuff after this change and the deployment?

**Comment by brianquinlan** [comment]
I think that it would be safe to remove the GKE stuff.

File: badge_server/README.rst [hunk 4]
```diff
@@ -1,7 +1,9 @@
 Badge Server
 ============
 
-    Displaying the compatibility status for your package as a Github Badge.
+.. image:: http://35.226.8.89/one_badge_image?package=tensorflow
```
Comment (liyanhui1228) [inline badge_server/README.rst]: Is this badge image added accidentally?

**Pull request merged**